Makefile

```diff
@@ -1,58 +1,178 @@
+# Basic variables
 GOOS 					?= $(shell go env GOOS)
 GOARCH 					?= $(shell go env GOARCH)
 VERSION 				?= $(shell hack/version.sh)
-# Images management
-REGISTRY				?= "docker.io/karmada"
-REGISTRY_USER_NAME  	?= ""
-REGISTRY_PASSWORD   	?= ""
-REGISTRY_SERVER_ADDRESS ?= ""
 
+# Default target when just running 'make'
+.DEFAULT_GOAL := all
+
+# Build targets
 TARGETS := karmada-dashboard-api \
-		   karmada-dashboard-web \
-
-
-# Build binary.
-#
-# Args:
-#   GOOS:   OS to build.
-#   GOARCH: Arch to build.
-#
-# Example:
-#   make
-#   make all
-#   make karmada-dashboard-api GOOS=linux
+		   karmada-dashboard-web
+
+# Docker image related variables
+REGISTRY				?= docker.io/karmada
+REGISTRY_USER_NAME  	?= 
+REGISTRY_PASSWORD   	?= 
+REGISTRY_SERVER_ADDRESS ?= 
+IMAGE_TARGET := $(addprefix image-, $(TARGETS))
+
+# Development server variables
+KARMADA_CTX ?= karmada-apiserver
+HOST_CTX ?= karmada-host
+API_PORT ?= 8000
+API_HOST ?= http://localhost:8000
+SKIP_TLS_VERIFY ?= false
+KUBECONFIG ?= $(HOME)/.kube/karmada.config
+
+###################
+# Build Targets   #
+###################
+
+# Build all binaries (alias for build)
+.PHONY: all
+all: build
+
+# Build all binaries
+.PHONY: build
+build: $(TARGETS)
+
+# Build specific binary
 .PHONY: $(TARGETS)
 $(TARGETS):
 	BUILD_PLATFORMS=$(GOOS)/$(GOARCH) hack/build.sh $@
 
-.PHONY: all
-all: $(TARGETS)
-
-# Build image.
-#
-# Args:
-#   GOARCH:      Arch to build.
-#   OUTPUT_TYPE: Destination to save image(docker/registry).
-#
-# Example:
-#   make images
-#   make image-karmada-dashboard-api
-#   make image-karmada-dashboard-api GOARCH=arm64
-IMAGE_TARGET=$(addprefix image-, $(TARGETS))
+###################
+# Docker Images   #
+###################
+
+# Build all images
+.PHONY: images
+images: $(IMAGE_TARGET)
+
+# Build specific image
 .PHONY: $(IMAGE_TARGET)
 $(IMAGE_TARGET):
 	set -e;\
 	target=$$(echo $(subst image-,,$@));\
 	make $$target GOOS=linux;\
 	VERSION=$(VERSION) REGISTRY=$(REGISTRY) BUILD_PLATFORMS=linux/$(GOARCH) hack/docker.sh $$target
 
+###################
+# UI Building     #
+###################
 
+.PHONY: bundle-ui-dashboard
 bundle-ui-dashboard:
 	cd ui && pnpm run dashboard:build
-bin-karmada-dashboard-web:
-	BUILD_PLATFORMS=$(GOOS)/$(GOARCH) hack/build.sh karmada-dashboard-web
-image-karmada-dashboard-web:
-	BUILD_PLATFORMS=linux/$(GOARCH) hack/build.sh karmada-dashboard-web
-	cp -R ui/apps/dashboard/dist _output/bin/linux/$(GOARCH)/dist
-	DOCKER_FILE="build-web.Dockerfile" VERSION=$(VERSION) REGISTRY=$(REGISTRY) BUILD_PLATFORMS=linux/$(GOARCH) hack/docker.sh karmada-dashboard-web
```

Comment:
This change breaks [the CI workflow](https://github.com/karmada-io/dashboard/actions/runs/12864209071/job/35862322142):
```
  make bin-***-dashboard-web
  shell: /usr/bin/bash -e {0}
  env:
    IMAGE_NAME: ***/***-dashboard-web
    BINARY_NAME: ***-dashboard-web
make: *** No rule to make target 'bin-***-dashboard-web'.  Stop.
Error: Process completed with exit code 2.
```

Reply:
I'll send a PR to fix it.